src/main/java/com/health/threadpool/TaskProcessingService.java

```diff
@@ -355,7 +356,7 @@ public void addUpdateDeleteTutorial(Tutorial tutorial, String requestType) {
 
         }
 
-        else if (tutorial.isAddedQueue()) {
+        else if (tutorial.isAddedQueue() && isURLWorking(documentPathforOriginalScript)) {
```

Comment:
don't call isURLWorking twice for the same URL. assign to a boolean variable once and use it.